pandas/tests/indexing/test_datetime.py

```diff
@@ -123,6 +125,74 @@ def test_consistency_with_tz_aware_scalar(self):
         result = df[0].at[0]
         assert result == expected
 
+    def test_access_timezoned_datetimeindex_with_timezoned_label(self):
+
+        # GH 6785
+        # timezone was ignored when string was provided as a label
+
+        first_january = pd.date_range('2016-01-01T00:00', '2016-01-01T23:59',
+                                      freq='T', tz="UTC")
+        df = pd.DataFrame(index=first_january, data=np.arange(len(
+                          first_january)))
+
+        former_naive_endpoint_df = df[
+            "2016-01-01T00:00-02:00":"2016-01-01T02:03"
+        ]
+
+        former_non_naive_endpoint_df = df[
```

Comment:
construct the expected return dataframe and simply compare using ``assert_frame_equal``

Reply:
thanks for that hint!